src/components/Table/types.ts

```diff
@@ -0,0 +1,142 @@
+
+export type ColumnType<T> = {
+  id: number;
+  name: string;
+  label?: string;
+  hidden?: boolean;
+  sort?: ((a: RowType<T>, b: RowType<T>) => number) | undefined;
+  render?: ({ value, row }: RenderFunctionArgsType<T>) => React.ReactNode;
+  headerRender?: HeaderRenderType<T>;
+};
+
+export type ColumnStateType<T> = {
+  id: number;
+  name: string;
+  label: string;
+  hidden: boolean;
+  sort?: ((a: RowType<T>, b: RowType<T>) => number) | undefined;
+  sorted: {
+    on: boolean;
+    asc?: boolean;
+  };
+  headerRender?: HeaderRenderType<T>;
+};
+
+export type HeaderRenderType<T> = ({ label }: { label: T }) => React.ReactNode;
+
+// this is the type saved as state and returned
+export type HeaderType<T> = {
+  id: number;
+  name: string;
+  label?: string;
+  hidden?: boolean;
+  sorted: {
+    on: boolean;
+    asc?: boolean;
+  };
+  sort?: ((a: RowType<T>, b: RowType<T>) => number) | undefined;
+  render: () => React.ReactNode;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type DataType = { [key: string]: any };
+
+export type ColumnByNamesType<T> = {
+  [key: string]: ColumnType<T>;
+};
+
+export type RenderFunctionType<T> = ({ value, row }: RenderFunctionArgsType<T>) => React.ReactNode | undefined;
+
+type RenderFunctionArgsType<T> = {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any;
+  row: T;
+};
+
+export type ColumnByNameType<T> = Omit<Required<ColumnType<T>>, "name" | "sort">;
+
+export interface RowType<T extends DataType> {
+  id: number;
+  cells: CellType[];
+  hidden?: boolean;
+  selected?: boolean;
+  original: T;
+}
+
+export type CellType = {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any;
```

Comment:
The value is a `ReactNode` is it not?

Reply:
Right

Reply:
A string is also a ReactNode.

Reply:
Yeah, my explanation was wrong. ReactNode includes string and number but value doesn't accept number. I will update it